pkg/collector/horizontalpodautoscaler.go

```diff
@@ -86,11 +86,17 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 			},
 		}
 
-		if otelcol.Spec.Autoscaler != nil && otelcol.Spec.Autoscaler.Behavior != nil {
+		if otelcol.Spec.Autoscaler.Behavior != nil {
 			behavior := ConvertToV2beta2Behavior(*otelcol.Spec.Autoscaler.Behavior)
 			autoscaler.Spec.Behavior = &behavior
 		}
 
+		// check for custom metrics
+		if len(otelcol.Spec.Autoscaler.Metrics) > 0 {
+			metrics := ConvertToV2Beta2PodMetrics(otelcol.Spec.Autoscaler.Metrics)
```

Comment:
are we able to make this addition to confirm to the v2 spec to start with? Or given it's already using something similar to the v1alpha we want to keep it that way?

Reply:
hm maybe i'm not understanding what this should be doing. What's the type of `autoscaler.Spec.Metrics` and how does that diff from what you're adding in?

Reply:
`otelcol.Spec.Autoscaler.Metrics` has type `v1alpha1.MetricSpec` this supports only the `PodsMetricSource`

Reply:
The HPA is either expecting a `v2.MetricSpec` or a `v2beta2.MetricSpec` so this does the necessary conversion for the HPA object depending on the version. This is something similar done for `.Spec.Autoscaler.Behavior`

Reply:
The reason I created `v1alpha1.MetricSpec` instead of directly using a `v2.MetricSpec` is because `v2.MetricSpec` is way too large of an object for the use cases of our HPA (see https://pkg.go.dev/k8s.io/api/autoscaling/v2#MetricSpec). It supports 5 different metric types which would require a lot of validation and the deprecation of `targetCPUUtilization` and `targetMemoryUtilization`.

Reply:
`v1alpha1.MetricSpec` is much smaller and can be updated as more functionality is required. I'm totally open to suggestions to change the type of `otelcol.Spec.Autoscaler.Metrics` though if that's still unclear!

Reply:
nope, makes a ton of sense. thanks for explaining!